core/src/main/scala/org/apache/spark/status/api/v1/ApiRootResource.scala

```diff
@@ -83,6 +83,8 @@ private[spark] trait UIRoot {
   def getApplicationInfoList: Iterator[ApplicationInfo]
   def getApplicationInfo(appId: String): Option[ApplicationInfo]
 
+  def executorThreadDumpsNotAvailableError(): Option[Response] = None
```

Comment:
This feels a little odd. You could instead have a `def threadDump(...): Response` method and have the implementation of it be in each `UIRoot` implementation, instead of in the resource class. I think I prefer that a little bit more over this.

Reply:
UIRoot has no access to AppStatusStore and SparkContext can I pass this parameters along with the executorId to the UIRoot?

Reply:
Sure, the method you'd add could have the `SparkUI` as a parameter.

Reply:
In fact, now that you pointed out the method to get the thread dump is in `SparkContext`, maybe you don't even need this at all. In the existing resource code, you can just return the error if `ui.sc` is not defined (e.g. "Thread dumps are only available in live applications.")